### ISSUE TITLE
pip install fann2 fails

### ISSUE THREAD

**Issue by rencoder** [FutureLinkCorporation/fann2#9]
_Log_

------------------------------------------------------------
>C:\Users\Rencoder\Anaconda\Scripts\pip-script.py run on 05/03/15 17:44:32
Downloading/unpacking fann2
  Getting page https://pypi.python.org/simple/fann2/
  URLs to search for versions for fann2:
  * https://pypi.python.org/simple/fann2/
  Analyzing links from page https://pypi.python.org/simple/fann2/
    Found link https://pypi.python.org/packages/source/f/fann2/fann2-1.0.0.tar.gz#md5=b0db8db52ced14067c7ec143a269fb8b (from https://pypi.python.org/simple/fann2/), version: 1.0.0
    Found link https://pypi.python.org/packages/source/f/fann2/fann2-1.0.7.tar.gz#md5=9ea834fba950ab0f6113822bb168330f (from https://pypi.python.org/simple/fann2/), version: 1.0.7
  Using version 1.0.7 (newest of versions: 1.0.7, 1.0.0)
  Downloading from URL https://pypi.python.org/packages/source/f/fann2/fann2-1.0.7.tar.gz#md5=9ea834fba950ab0f6113822bb168330f (from https://pypi.python.org/simple/fann2/)
  Running setup.py (path:c:\users\Rencoder\appdata\local\temp\pip_build_Rencoder\fann2\setup.py) egg_info for package fann2
    running swig
    running egg_info
    creating pip-egg-info\fann2.egg-info
    writing pip-egg-info\fann2.egg-info\PKG-INFO
    writing top-level names to pip-egg-info\fann2.egg-info\top_level.txt
    writing dependency_links to pip-egg-info\fann2.egg-info\dependency_links.txt
    writing manifest file 'pip-egg-info\fann2.egg-info\SOURCES.txt'
    warning: manifest_maker: standard file '-c' not found
    
>reading manifest file 'pip-egg-info\fann2.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pip-egg-info\fann2.egg-info\SOURCES.txt'
  Source in c:\users\Rencoder\appdata\local\temp\pip_build_Rencoder\fann2 has version 1.0.7, which satisfies requirement fann2
Installing collected packages: fann2
  Running setup.py install for fann2
    Running command C:\Users\Rencoder\Anaconda\python.exe -c "import setuptools, tokenize;__file__='c:\\users\\Rencoder\\appdata\\local\\temp\\pip_build_Rencoder\\fann2\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\Rencoder\appdata\local\temp\pip-yokqci-record\install-record.txt --single-version-externally-managed --compile
    running swig
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-2.7
    creating build\lib.win-amd64-2.7\fann2
    copying fann2\__init__.py -> build\lib.win-amd64-2.7\fann2
    copying fann2\libfann.py -> build\lib.win-amd64-2.7\fann2
    running egg_info
    writing fann2.egg-info\PKG-INFO
    writing top-level names to fann2.egg-info\top_level.txt
    writing dependency_links to fann2.egg-info\dependency_links.txt
    warning: manifest_maker: standard file '-c' not found
    
>    reading manifest file 'fann2.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'fann2.egg-info\SOURCES.txt'
    copying fann2\fann2.i -> build\lib.win-amd64-2.7\fann2
    copying fann2\fann2_wrap.cxx -> build\lib.win-amd64-2.7\fann2
    copying fann2\fann_cpp_subclass.h -> build\lib.win-amd64-2.7\fann2
    running build_ext
    building 'fann2._libfann' extension
    creating build\temp.win-amd64-2.7
    creating build\temp.win-amd64-2.7\Release
    creating build\temp.win-amd64-2.7\Release\fann2
    C:\Users\Rencoder\Anaconda\Scripts\gcc.bat -DMS_WIN64 -mdll -O -Wall -DSWIG_COMPILE -I./include -I../include -Iinclude -IC:\Users\Rencoder\Anaconda\include -IC:\Users\Rencoder\Anaconda\PC -c fann2/fann2_wrap.cxx -o build\temp.win-amd64-2.7\Release\fann2\fann2_wrap.o
    writing build\temp.win-amd64-2.7\Release\fann2\_libfann.def
    C:\Users\Rencoder\Anaconda\Scripts\g++.bat -DMS_WIN64 -shared -s build\temp.win-amd64-2.7\Release\fann2\fann2_wrap.o build\temp.win-amd64-2.7\Release\fann2\_libfann.def -LC:\Users\Rencoder\Anaconda\libs -LC:\Users\Rencoder\Anaconda\PCbuild\amd64 -ldoublefann -lpython27 -lmsvcr90 -o build\lib.win-amd64-2.7\fann2\_libfann.pyd
    c:/users/Rencoder/anaconda/mingw/bin/../lib/gcc/x86_64-w64-mingw32/4.7.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ldoublefann
    collect2.exe: error: ld returned 1 exit status
    error: command 'C:\\Users\\Rencoder\\Anaconda\\Scripts\\g++.bat' failed with exit status 1
    Complete output from command C:\Users\Rencoder\Anaconda\python.exe -c "import setuptools, tokenize;__file__='c:\\users\\Rencoder\\appdata\\local\\temp\\pip_build_Rencoder\\fann2\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\Rencoder\appdata\local\temp\pip-yokqci-record\install-record.txt --single-version-externally-managed --compile:
    running swig

> running install

>running build

>running build_py

>creating build

>creating build\lib.win-amd64-2.7

creating build\lib.win-amd64-2.7\fann2

copying fann2\__init__.py -> build\lib.win-amd64-2.7\fann2

copying fann2\libfann.py -> build\lib.win-amd64-2.7\fann2

running egg_info

writing fann2.egg-info\PKG-INFO

writing top-level names to fann2.egg-info\top_level.txt

writing dependency_links to fann2.egg-info\dependency_links.txt

warning: manifest_maker: standard file '-c' not found

reading manifest file 'fann2.egg-info\SOURCES.txt'

reading manifest template 'MANIFEST.in'

writing manifest file 'fann2.egg-info\SOURCES.txt'

copying fann2\fann2.i -> build\lib.win-amd64-2.7\fann2

copying fann2\fann2_wrap.cxx -> build\lib.win-amd64-2.7\fann2

copying fann2\fann_cpp_subclass.h -> build\lib.win-amd64-2.7\fann2

running build_ext

building 'fann2._libfann' extension

creating build\temp.win-amd64-2.7

creating build\temp.win-amd64-2.7\Release

creating build\temp.win-amd64-2.7\Release\fann2

C:\Users\Rencoder\Anaconda\Scripts\gcc.bat -DMS_WIN64 -mdll -O -Wall -DSWIG_COMPILE -I./include -I../include -Iinclude -IC:\Users\Rencoder\Anaconda\include -IC:\Users\Rencoder\Anaconda\PC -c fann2/fann2_wrap.cxx -o build\temp.win-amd64-2.7\Release\fann2\fann2_wrap.o

writing build\temp.win-amd64-2.7\Release\fann2\_libfann.def

C:\Users\Rencoder\Anaconda\Scripts\g++.bat -DMS_WIN64 -shared -s build\temp.win-amd64-2.7\Release\fann2\fann2_wrap.o build\temp.win-amd64-2.7\Release\fann2\_libfann.def -LC:\Users\Rencoder\Anaconda\libs -LC:\Users\Rencoder\Anaconda\PCbuild\amd64 -ldoublefann -lpython27 -lmsvcr90 -o build\lib.win-amd64-2.7\fann2\_libfann.pyd

c:/users/Rencoder/anaconda/mingw/bin/../lib/gcc/x86_64-w64-mingw32/4.7.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ldoublefann

collect2.exe: error: ld returned 1 exit status

error: command 'C:\\Users\\Rencoder\\Anaconda\\Scripts\\g++.bat' failed with exit status 1

----------------------------------------
Cleaning up...
  Removing temporary dir c:\users\Rencoder\appdata\local\temp\pip_build_Rencoder...
Command C:\Users\Rencoder\Anaconda\python.exe -c "import setuptools, tokenize;__file__='c:\\users\\Rencoder\\appdata\\local\\temp\\pip_build_Rencoder\\fann2\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\Rencoder\appdata\local\temp\pip-yokqci-record\install-record.txt --single-version-externally-managed --compile failed with error code 1 in c:\users\Rencoder\appdata\local\temp\pip_build_Rencoder\fann2
Exception information:
Traceback (most recent call last):
  File "C:\Users\Rencoder\Anaconda\lib\site-packages\pip\basecommand.py", line 122, in main
    status = self.run(options, args)
  File "C:\Users\Rencoder\Anaconda\lib\site-packages\pip\commands\install.py", line 283, in run
    requirement_set.install(install_options, global_options, root=options.root_path)
  File "C:\Users\Rencoder\Anaconda\lib\site-packages\pip\req.py", line 1435, in install
    requirement.install(install_options, global_options, *args, **kwargs)
  File "C:\Users\Rencoder\Anaconda\lib\site-packages\pip\req.py", line 706, in install
    cwd=self.source_dir, filter_stdout=self._filter_install, show_stdout=False)
  File "C:\Users\Rencoder\Anaconda\lib\site-packages\pip\util.py", line 697, in call_subprocess
    % (command_desc, proc.returncode, cwd))
>InstallationError: Command C:\Users\Rencoder\Anaconda\python.exe -c "import setuptools, tokenize;__file__='c:\\users\\Rencoder\\appdata\\local\\temp\\pip_build_Rencoder\\fann2\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\Rencoder\appdata\local\temp\pip-yokqci-record\install-record.txt --single-version-externally-managed --compile failed with error code 1 in c:\users\Rencoder\appdata\local\temp\pip_build_Rencoder\fann2

**Comment by FutureLinkOpenSource** [comment]
Hi, did you ensure the fann library was installed before installing py-fann2? Please, let us know that it was installed, that you cleaned your build environment, and reinstalled py-fann2. If it's failing after that, let's see how we can help.

**Comment by rencoder** [comment]
I appreciate the quick response. I downloaded [this](http://sourceforge.net/projects/fann/files/fann/2.2.0/FANN-2.2.0-Source.zip/download) source and built the Visual Studio project as instructed [here](http://leenissen.dk/fann/wp/help/installing-fann/). It built successfully and I could find EXEs and DLLs in the bin folder. Thereafter, I ran the pip install command but it failed.

**Comment by tty02-fl** [comment]
The *EXE* and *DLL* in bin folder should be accessible widely in the system  or at minimum to python and SWIG software before installing py-fann2.

**Comment by rencoder** [comment]
Okay, I managed to solve it. Your advice was key, @tty02-fl; thank you very much.

I examined the -L parameter of g++.bat (that executes during `pip install fann2`) based on [this](http://ru.stackoverflow.com/a/129653) Stackoverflow answer. Its value was `C:\Users\Rencoder\Anaconda\libs`, which is where it tried (and failed) to find the _doublefann_ object file library. Should it also have searched the folders under path variable automatically?

Anyways, I manually copied the _doublefann_ object file library from my Visual Studio Release build to the Anaconda libs folder. I also had to rename the file to _doublefann_ as the Visual Studio build outputted it as _fanndouble_. This time the installer mentioned something about _doublefann_ file being incompatible. So I rebuilt the Visual Studio project with x64 solution settings instead of the default Win32 settings and replaced the object file library with the new x64 one.

`pip install` now succeeded. Thanks again, @FutureLinkOpenSource and @tty02-fl for your quick response and help with this.
